### PR TITLE
CDAP-13099 add cloud runtime walk through and screenshots

### PR DESCRIPTION
**Pull request by albertshau**

### CODE REVIEW

**Comment by albertshau** [comment]
https://builds.cask.co/browse/CDAP-DQB422

File: cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst [hunk 1]
```diff
@@ -10,5 +10,125 @@
 Example Walk-Through
 ====================
 
-Insert step by step instructions for creating, assigning, monitoring, disabling, deleting.
+This example walks you through creating a profile using the Google Dataproc provisioner,
```
Comment (deb-saunders) [inline cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst]: >  Google Dataproc provisioner, then
Change to:
 Google Dataproc provisioner and then

File: cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst [hunk 2]
```diff
@@ -10,5 +10,125 @@
...
+then running a pipeline on a Google Dataproc cluster using the new profile.
+
+Setup
+-----
+
+The Google Dataproc provisioner requires that you perform a little bit of setup on your Google Cloud Platform (GCP) account.
+All of these steps can be performed in the GCP Console.
```
Comment (deb-saunders) [inline cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst]: You can perform all these steps in the GCP Console.

File: cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst [hunk 3]
```diff
@@ -10,5 +10,125 @@
...
+
+  - Enable Cloud Dataproc
+  - Create a service account key that has permission to create Cloud Dataproc clusters and view Google Compute Engine resources
+  - Ensure there is an ingress VPC firewall rule that allows tcp connections on port 443 (https)
```
Comment (deb-saunders) [inline cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst]: tcp > TCP
https > HTTPS

File: cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst [hunk 4]
```diff
@@ -10,5 +10,125 @@
...
+  - Create a Google Cloud Storage (GCS) bucket and upload a test file
```
Comment (deb-saunders) [inline cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst]: For branding
Google Cloud Storage 1st mention
Cloud Storage subsequent mention
not GCS

File: cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst [hunk 5]
```diff
@@ -10,5 +10,125 @@
...
+
+After you have set up your GCP account, add the service account key as a key in the CDAP secure store.
```
Comment (deb-saunders) [inline cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst]: +After you have set up your GCP account, add the service account key as a key in the CDAP secure store.
+This must be done using the :ref:`Secure Storage RESTful API <http-restful-api-secure-storage>`.
Change to:
After setting up your GCP account, use the :ref:`Secure Storage RESTful API <http-restful-api-secure-storage>` to add the service account key as a key in the CDAP secure store.

File: cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst [hunk 6]
```diff
@@ -10,5 +10,125 @@
...
+
+Select the Google Cloud Dataproc provisioner.
+
+.. figure:: /_images/cloud-runtimes/provisioner-configure.png
+  :figwidth: 100%
+  :width: 800px
+  :align: center
+  :class: bordered-image
+
+Fill in all configuration fields. For the service account key, specify the name of the secure key that
```
Comment (deb-saunders) [inline cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst]: Change to:
For the service account key, specify the name of the key that you added during setup.

File: cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst [hunk 7]
```diff
@@ -10,5 +10,125 @@
...
+was added in the setup part of this walk-through.
+
+Create a Test Pipeline
+----------------------
+
+Click on the green plus button and create a new batch pipeline. Our test pipeline will be a very simple
```
Comment (deb-saunders) [inline cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst]: Change to: 
This simple test pipeline reads a file in Cloud Storage and writes to an output directory in Cloud Storage.

File: cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst [hunk 8]
```diff
@@ -10,5 +10,125 @@
...
+To create a profile, navigate to the Administration page using the menu at the top right of the CDAP UI.
+From this page, you can view the list of system profiles.
+
+.. figure:: /_images/cloud-runtimes/create-system-profile.png
+  :figwidth: 100%
+  :width: 800px
+  :align: center
+  :class: bordered-image
+
+Click on the button to create a new profile.
```
Comment (deb-saunders) [inline cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst]: omit "on"
Click the button...

File: cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst (hunk 6 above)
Comment (deb-saunders) [inline cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst]: Add "the"
Fill in all the configuration ....

File: cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst (hunk 7 above)
Comment (deb-saunders) [inline cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst]: Omit "on"
Click the green ...

File: cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst [hunk 9]
```diff
@@ -10,5 +10,125 @@
...
+one that simply reads a file on GCS and writes to an output directory on GCS. Begin by adding a File source
+and a File sink from the left panel. Connect the source to the sink.
+
+.. figure:: /_images/cloud-runtimes/simple-pipeline-overview.png
+  :figwidth: 100%
+  :width: 800px
+  :align: center
+  :class: bordered-image
+
+Configure the source to read a file from the GCS bucket that you set up earlier.
```
Comment (deb-saunders) [inline cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst]: Is this the test file that the user was supposed to upload during setup? 
Change to:
Configure the source to read the test file from the Cloud Storage bucket that you created during setup.

File: cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst [hunk 10]
```diff
@@ -10,5 +10,125 @@
...
+
+Configure the sink to write to a directory that does not yet exist.
+
+.. figure:: /_images/cloud-runtimes/simple-pipeline-output.png
+  :figwidth: 100%
+  :width: 800px
+  :align: center
+  :class: bordered-image
+
+Deploy the pipeline by clicking on the `Deploy` button near the top right of the screen.
```
Comment (deb-saunders) [inline cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst]: Change to:
To deploy the pipeline, click the `Deploy` button near the top right of the screen, which opens the pipeline overview page. In this page, configure the pipeline to use the profile that you created earlier.

**Comment by albertshau** [comment]
https://builds.cask.co/artifact/CDAP-DQB422/shared/build-1/Docs-HTML/html/developer-manual/cloud-runtimes/index.html

File: cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst [hunk 11]
```diff
@@ -10,5 +10,125 @@
...
+
+Note that you can also override any profile setting that was not locked during profile creation.
+
+.. figure:: /_images/cloud-runtimes/pipeline-profile-override.png
+  :figwidth: 100%
+  :width: 800px
+  :align: center
+  :class: bordered-image
+
+Run the pipeline. The pipeline should transition to the `Pending` status while the Cloud Dataproc
```
Comment (deb-saunders) [inline cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst]: Run the pipeline.  
On the Cloud Dataproc page in the GCP Console, you can monitor the progress of the pipeline. `Pending` means Cloud Dataproc is provisioning the cluster for the run. When completed, the pipeline status transitions to `Starting` > `Running` > `Succeeded`.

File: cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst [hunk 12]
```diff
@@ -10,5 +10,125 @@
...
+cluster is being provisioned for the run. You can view the Cloud Dataproc section of the GCP Console
+to track progress of the cluster. After the cluster has been provisioned, you will see the pipeline
+status transition to `Starting`, then `Running`, then `Succeeded`. This, of course, assumes that your
+GCP account and your pipeline are properly configured.
```
Comment (deb-saunders) [inline cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst]: unless you're going to add troubleshooting info, suggest we omit:
This, of course, assumes that your GCP account and your pipeline are properly configured.

File: cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst (hunk 11 above)
Comment (deb-saunders) [reply cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst]: Is this the Cloud Dataproc Jobs page or the Cluster page that shows this information?

File: cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst [hunk 13]
```diff
@@ -10,5 +10,125 @@
...
+
+After the pipeline has run, navigate back to the Administration page and click on your profile to view
```
Comment (deb-saunders) [inline cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst]: change to:
After the pipeline runs, navigate to the Administration page and click your profile to view its details.

File: cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst [hunk 14]
```diff
@@ -10,5 +10,125 @@
...
+details about the profile.
+
+.. figure:: /_images/cloud-runtimes/profile-overview.png
+  :figwidth: 100%
+  :width: 800px
+  :align: center
+  :class: bordered-image
+
+From this page, you can see usage metrics for the profile, as well as what pipeline have been assigned to it.
```
Comment (deb-saunders) [inline cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst]: Change to:
The Administration pages shows you usage metrics for the profile and its assigned pipelines. 

You have now successfully created and used a profile to execute a pipeline in a cloud environment.

**Review by deb-saunders — CHANGES_REQUESTED** [review]

File: cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst (hunk 9 above)
Comment (albertshau) [reply cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst]: yeah that's correct

File: cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst (hunk 11 above)
Comment (albertshau) [reply cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst]: Pending > Starting > Running > Succeeded is the CDAP pipeline UI. The Clusters page for Cloud Dataproc will have Provisioning > Running > Deleting

Comment (albertshau) [reply cdap-docs/developer-manual/source/cloud-runtimes/example/index.rst]: I'm going to remove the part about looking at the Dataproc UI and only talk about the CDAP UI. It's too confusing to mention two UIs.

**Comment by albertshau** [comment]
https://builds.cask.co/artifact/CDAP-DQB422/shared/build-2/Docs-HTML/html/developer-manual/cloud-runtimes/index.html

**Pull request merged**